pkg/prometheus/promcfg.go

```diff
@@ -844,6 +873,21 @@ func (cg *configGenerator) generateServiceMonitorConfig(
 	return cfg
 }
 
+func enforceNamespaceLabel(namespace, enforcedNamespaceLabel string) (relabeling yaml.MapSlice) {
```

Comment:
I think it would be a bit more natural not to do a named return here and pass the existing relabeling configs in here and append the label enforcing instead of having to deal with a default value return value.